var/spack/repos/builtin/packages/armadillo/package.py

```diff
@@ -13,9 +13,12 @@ class Armadillo(CMakePackage):
 
     homepage = "https://arma.sourceforge.net/"
     url = "http://sourceforge.net/projects/arma/files/armadillo-8.100.1.tar.xz"
+    git = "https://gitlab.com/conradsnicta/armadillo-code"
```

Comment:
You want to add the extension:

```suggestion
    git = "https://gitlab.com/conradsnicta/armadillo-code.git"
```

Reply:
Are you planning to add branch versions?

Reply:
No, but when adding new versions the git url is useful to get a diff with previous versions. I tend to add it when it is not trivially easy based on homepage or url.